_LowLevelCode/cpp/file_parameters/fileParameters.cpp

```diff
@@ -0,0 +1,185 @@
+const std::string fileParameters::MEX_ERR_ID("HORACE:fileParameters:invalid_argument");
+//--------------------------------------------------------------------------------------------------------------------
+//--------------------------------------------------------------------------------------------------------------------
+//--------------------------------------------------------------------------------------------------------------------
+/* return input parameters as cellarray of character arrays
+*  Function used in testing fileParameters interface to Matlab
+*/
+void fileParameters ::returnInputs(mxArray** const outParList) {
+    auto out = mxCreateCellMatrix(this->num_input_params,1);
+    outParList[0] = out;
+    int n_set_param(0); // number of parameter set as the result
+    for (size_t i = 0; i < fileParamNames.size(); i++) {
+        if (!this->parameters_set[i])continue;
+        switch (i) {
+        case(0): {
+            auto pFn = mxCreateString(this->fileName.c_str());
+            mxSetCell(out, n_set_param, pFn);
+            break;
+        }
+        case(1): {
+			auto pVal = mxCreateDoubleScalar(double(this->nbin_start_pos));
+			mxSetCell(out, n_set_param, pVal);
+            break;
+        }
+        case(2): {
+            auto pVal = mxCreateDoubleScalar(double(this->pix_start_pos));
+            mxSetCell(out, n_set_param, pVal);
+            break;
+        }
+        case(3): {
+			auto pVal = mxCreateDoubleScalar(double(this->run_id));
+			mxSetCell(out, n_set_param, pVal);
+            break;
+        }
+        case(4): {
+			auto pVal = mxCreateDoubleScalar(double(this->total_NfileBins));
+			mxSetCell(out, n_set_param, pVal);
+            break;
+        }
+        case(5): {
+			auto pVal = mxCreateDoubleScalar(double(this->total_nPixels));
+			mxSetCell(out, n_set_param, pVal);
+            break;
+        }
+        case(6): {
+			auto pVal = mxCreateDoubleScalar(double(this->pixel_width));
+			mxSetCell(out, n_set_param, pVal);
+            break;
+        }
+        default: {
+            mexWarnMsgTxt("combine_sqw: unknown parameter (should never happen)");
+        }
+        };
+        n_set_param++;
+    };
+}
+/* Convert input Matlab structure with the fields, necessary to describe input & output sqw file into
+ * fileParameters class.
+ @input -- pointer to Matlab structure, containing the file description, with fields defined in the map above.
+*/
+fileParameters::fileParameters(const mxArray* pFileParam) :
+   fileParameters()
+{
+
+    mwSize total_num_of_elements = mxGetNumberOfElements(pFileParam);
+    mwSize number_of_fields = mxGetNumberOfFields(pFileParam);
+
+    if (total_num_of_elements != 1) {
+        std::stringstream buf;
+        buf << "each field of file parameter structure should contain only one element, not: " << (short)total_num_of_elements << std::endl;
+        mexErrMsgIdAndTxt(MEX_ERR_ID.c_str(),buf.str().c_str());
```

Comment:
because sprintf will print result into stdout (it certainly will not be desplayed in Matlab screen) and pattern used here will return exception with standard Matlab structure `HORACE:code:reason,"details of the issue"`

as for endl -- very interesting discussion but it should be `std::endl` exactly for the reasons described there. I am building exception and do not want its info it to stack  somewhere in buffers.